receiver/splunkhecreceiver/receiver.go

```diff
@@ -208,38 +241,66 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
 			return
 		}
 		if msg.IsMetric() {
-			// Currently unsupported.
-			r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, err)
+			if r.metricsConsumer == nil {
+				r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedMetricEvent, err)
+				return
+			}
+		} else if r.logsConsumer == nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errUnsupportedLogEvent, err)
 			return
 		}
 
-		messagesReceived++
 		events = append(events, &msg)
 	}
+	if r.logsConsumer != nil {
+		r.consumeLogs(ctx, events, resp, req)
+	} else {
+		r.consumeMetrics(ctx, events, resp, req)
+	}
```

Comment:
Is it possible for logsConsumer and metricsConsumer t be non-nil at the same time?

Reply:
We discussed whether this receiver should be able to simultaneously support logs and metrics with the same server (e.g. the global map thing), but it was determined that for the initial receiver it is unnecessary.

Reply:
Can you please clarify the rationale for the decision? Other receivers support this, there is a known way to implement it and it is a desirable capability.

Reply:
Yes, there are a number of reasons and subtleties that force a bit of a ham-handed simplistic approach here, that I would like to see refined over time.
1. (and probably the best reason) in the past, when I sent PRs that were a bit (very) long, I was asked to break them up. I feel like this one is complex as it is - so I'd be in favor of adding a TODO in there and do this in a future PR.
2. Each Splunk event may have an index field which points to a metrics or event index. We currently do not support passing through the index field all the way to the exporter as a resource attribute. I was trying to get #1289 first in so it'd be a bit easier to add this on both ends and test end to end.
3. We should be ready to expect folks use the same endpoint for both metrics and logs. In that case, they might even send splunk events which are interleaved with logs and metrics at once. We can then at best support one log pipeline and one metric pipeline with one endpoint. I'm not sure how to go about that competently. It feels ever so slightly different from the approach SignalFx receivers took, which I would need to study some more.

Reply:
@atoulme please add a todo to look into this further. I'll approve as is.